src/v/kafka/server/request_context.h

```diff
@@ -152,6 +160,9 @@ class request_context {
           ResponseType::api_type::key,
           ResponseType::api_type::name,
           r);
+        if constexpr (has_throttle_member<ResponseType>::value) {
+            r.throttle_time_ms = throttle_delay_ms();
```

Comment:
nice use of sfinae!

I don't think `r` (aka `ResponseType`) will ever have the `throttle_time_ms` member given the way that `request_context::respond` is invoked, right?

Reply:
nice catch, I should be performing this check/set on a `data` member within `r`.